community/modules/container/artifact-registry/variables.tf

```diff
@@ -0,0 +1,148 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Todo: re-add the variable validation in a way that pre-commit/TF is happy with...
+
+variable "project_id" {
+  description = "Project ID where the artifact registry and secret are created."
+  type        = string
+}
+
+variable "region" {
+  description = "Region for the artifact registry."
+  type        = string
+}
+
+variable "deployment_name" {
+  description = "The name of the current deployment."
+  type        = string
+}
+
+variable "labels" {
+  description = "Labels to add to the artifact registry. Key-value pairs."
+  type        = map(string)
+  default     = {}
+}
+
+variable "repo_password" {
+  description = "Optional password/API key. If null, one will be randomly generated."
+  type        = string
+  default     = null
+
+  # To Do: implement validation
+  # validation {
+  #   condition     = (var.repo_password == null || (var.use_upstream_credentials == true && var.repo_mode == "REMOTE_REPOSITORY"))
+  #   error_message = "repo_password can only be set if repo_mode is REMOTE_REPOSITORY and use_upstream_credentials is true. Otherwise, leave it null."
+  # }
+}
+
+variable "user_managed_replication" {
+  description = <<-DOC
+    (Optional) A list of objects to enable user-managed replication.
+    Each object can have:
+      location        = string
+      kms_key_name    = optional(string)
+    If empty, auto replication is used.
+  DOC
+  type = list(object({
+    location     = string
+    kms_key_name = optional(string)
+  }))
+  default = []
+}
+
+variable "format" {
+  description = "Artifact Registry format (e.g., DOCKER)."
+  type        = string
+  default     = "DOCKER"
+}
+
+variable "repo_mode" {
+  description = "Artifact Registry mode (STANDARD_REPOSITORY, REMOTE_REPOSITORY, etc.)."
+  type        = string
+  default     = "STANDARD_REPOSITORY"
+
+  validation {
+    condition     = can(regex("^(STANDARD_REPOSITORY|REMOTE_REPOSITORY|VIRTUAL_REPOSITORY)$", var.repo_mode))
+    error_message = "repo_mode must be one of STANDARD_REPOSITORY, REMOTE_REPOSITORY, or VIRTUAL_REPOSITORY."
+  }
+}
+
+variable "repo_public_repository" {
+  description = <<-DOC
+    For REMOTE_REPOSITORY, name of a known public repo as per the Terraform module
+    (e.g., DOCKER_HUB) or null for custom repo.
+  DOC
+  type        = string
+  default     = null
+
+  # To Do: implement validation
+  # validation {
+  #   condition     = ((var.repo_mode != "REMOTE_REPOSITORY" && var.repo_public_repository == null) || (var.repo_mode == "REMOTE_REPOSITORY" && (var.repo_public_repository != null || var.repo_mirror_url != null)))
+  #   error_message = "If repo_mode is REMOTE_REPOSITORY, you must set either repo_public_repository or repo_mirror_url. Otherwise, leave them null."
+  # }
```

Comment:
Agreed. As written, this validation block would break for Terraform < 1.9, which breaks our guarantee of Terraform 1.5 compatibility.

The way to do this in Terraform 1.5 is via a lifecycle/precondition block:

https://developer.hashicorp.com/terraform/language/expressions/custom-conditions#preconditions-and-postconditions

```suggestion
```

Reply:
I am OK if we add the lifecycle/precondition at a later time.

Reply:
Code addressed this.